### PR TITLE
chore: update load library

### PR DESCRIPTION
**Pull request by AdamEECS**

### CODE REVIEW

**Review by huskar-t — APPROVED** [review]
LGTM

File: taos/cinterface.py [hunk 1]
```diff
@@ -30,17 +30,31 @@ class TaosOption:
 
 
 def _load_taos_linux():
-    return ctypes.CDLL("libtaos.so")
+    try:
+        return ctypes.CDLL("libtaos.so")
+    except Exception as e:
+        raise InterfaceError("unable to load taos C library: %s" % e)
```
Comment (sangshuduo) [inline taos/cinterface.py:36]: s/taos C/taos client

File: taos/cinterface.py [hunk 2]
```diff
@@ -30,17 +30,31 @@ class TaosOption:
...
 
 def _load_taos_darwin():
-    from ctypes.macholib.dyld import dyld_find as _dyld_find
+    try:
+        from ctypes.macholib.dyld import dyld_find as _dyld_find
 
-    return ctypes.CDLL(_dyld_find("libtaos.dylib"))
+        return ctypes.CDLL(_dyld_find("libtaos.dylib"))
+    except Exception as e:
+        raise InterfaceError("unable to load taos C library: %s" % e)
```
Comment (sangshuduo) [inline taos/cinterface.py:45]: s/taos C/taos client

File: taos/cinterface.py [hunk 3]
```diff
@@ -30,17 +30,31 @@ class TaosOption:
...
 
 
 def _load_taos_windows():
-    return ctypes.windll.LoadLibrary("taos")
+    try:
+        return ctypes.windll.LoadLibrary("taos")
+    except Exception as e:
+        print("unable to load taos C library: %s" % e)
```
Comment (sangshuduo) [inline taos/cinterface.py:52]: same here

File: taos/cinterface.py [hunk 4]
```diff
@@ -30,17 +30,31 @@ class TaosOption:
...
+        try:
+            from ctypes.util import find_library
+            ctypes.windll.LoadLibrary(find_library("taos"))
+        except Exception as final_err:
+            raise InterfaceError("unable to load taos C library: %s" % final_err)
```
Comment (sangshuduo) [inline taos/cinterface.py:57]: here

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comments

File: taos/cinterface.py (hunk 2 above)
Comment (AdamEECS) [reply taos/cinterface.py]: replaced by 'taos client'

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**